tests/templates/test_state_preparations/test_qrom_state_prep.py

```diff
@@ -0,0 +1,253 @@
+
+    @pytest.mark.parametrize(
+        ("state", "num_wires", "num_work_wires", "num_precision_wires"),
+        [
+            (np.array([0, 0, 0, 1]), 2, 5, 5),
+            (np.array([0, 0, 0, 1 / 2, 0, 1 / 2, 1 / 2, -1j / 2]), 3, 5, 5),
+            (np.array([-0.84223628, -0.40036496, 0.08974619, -0.34970212]), 2, 1, 6),
+            (
+                np.array(
+                    [
+                        0.17157142 + 0.09585932j,
+                        0.00852997 - 0.21056896j,
+                        0.12986199 + 0.94654822j,
+                        0.04206036 - 0.04873857j,
+                    ]
+                ),
+                2,
+                1,
+                7,
+            ),
+        ],
+    )
+    def test_correctness(self, state, num_wires, num_work_wires, num_precision_wires):
+
+        wires = qml.registers(
+            {"work": num_work_wires, "precision": num_precision_wires, "state": num_wires}
+        )
+
+        dev = qml.device("default.qubit", wires=num_work_wires + num_precision_wires + num_wires)
+
+        qs = qml.tape.QuantumScript(
+            qml.QROMStatePreparation.compute_decomposition(
+                state,
+                wires=wires["state"],
+                work_wires=wires["work"],
+                precision_wires=wires["precision"],
+            ),
+            [qml.state()],
+        )
+
+        program, _ = dev.preprocess()
+        tape = program([qs])
+        output = dev.execute(tape[0])[0][: 2**num_wires]
+
+        assert np.allclose(state, output, atol=0.05)
+
+    def test_decomposition(self):
+        """Test that the correct gates are added in the decomposition"""
+
+        wires = qml.registers({"work": 3, "precision": 3, "state": 2})
+
+        decomposition = qml.QROMStatePreparation.compute_decomposition(
+            np.array([1 / 2, 1j / 2, -1 / 2, -1j / 2]),
+            wires=wires["state"],
+            work_wires=wires["work"],
+            precision_wires=wires["precision"],
+        )
+
+        for gate in decomposition:
+            assert gate.name in ["QROM", "Adjoint(QROM)", "CRY", "C(GlobalPhase)"]
+
+    @pytest.mark.jax
+    def test_interface_jax(self):
+        """Test QROMStatePreparation works with jax"""
+
+        from jax import numpy as jnp
+
+        state = [1 / 2, -1 / 2, 1j / 2, -1j / 2]
+
+        wires = qml.registers({"work": 2, "precision": 2, "state": 2})
+        dev = qml.device("default.qubit", wires=6)
+
+        qs = qml.tape.QuantumScript(
+            qml.QROMStatePreparation.compute_decomposition(
+                jnp.array(state),
+                wires=wires["state"],
+                work_wires=wires["work"],
+                precision_wires=wires["precision"],
+            ),
+            [qml.state()],
+        )
+
+        program, _ = dev.preprocess()
+        tape = program([qs])
+        output_jax = dev.execute(tape[0])[0]
```

Comment:
We don't need to select the jax interface here ?

Reply:
Is the way we test interfaces different now? I remember before we would build a qnode and provide the interface explicitly there.

Reply:
Recently I was told to avoid using qnodes as much as possible. Do you think that this is one of those cases where it is necessary to?